[rocpd] Schema Updates

rocm-devops opened the pull request:
# PR Details

## Associated Jira Ticket Number/Link

## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Continuous Integration

## Technical details

<!-- Please explain the changes along with JIRA/Github link(if applies). -->

## Added/updated tests?

<!-- We encourage you to keep the code coverage percentage at 80% and above. -->

- [ ] Yes
- [ ] No, Does not apply to this PR.

## Updated CHANGELOG?

<!-- Needed for Release updates for a ROCm release. -->

- [ ] Yes
- [ ] No, Does not apply to this PR.

## Added/Updated documentation?

- [ ] Yes
- [ ] No, Does not apply to this PR.

jrmadsen committed:
Update rocprofiler-sdk-rocpd

- remove rocprofiler-sdk-rocpd marker_views.sql
- create rocpd_metadata.sql
- minor cmake updates
  - support multiple python versions by default

jrmadsen committed:
Misc minor updates

- output config annotate {args,pmc}

jrmadsen committed:
RocPD schema update

- rocpd_timestamp table
- expand the role of rocpd_track table
- fix some inconsistencies in tables and views
- reworked the perfetto implementation from the ground up
- support --annotate-args and --annotate-pmc in perfetto

jrmadsen committed:
SQL syntax error fix

jrmadsen committed:
rocpd command line tools + cmake updates

- ROCPROFILER_BUILD_PYTHON option to disable building python bindings
- various rocpd command-line tools to simplify invoking rocpd

jrmadsen committed:
Require ROCPROFILER_PYTHON_VERSIONS when ROCPROFILER_BUILD_PYTHON=ON

jrmadsen committed:
roctx annotation in mandelbrot benchmark

jrmadsen committed:
timer for extracting row count

jrmadsen committed:
Tweak group-by threads query in perfetto

jrmadsen committed:
Log-level for simple timer

jrmadsen committed:
Updates for simple timer

jrmadsen committed:
Improve rocpd perfetto timer logging

jrmadsen committed:
Fix simple timer logging

rocm-devops commented:
Manual review required for [e52d79](https://github.com/AMD-ROCm-Internal/rocprofiler-sdk-internal/runs/44732831661)
Assigned Auditors: @IAS

jayhawk-commits commented:
This pull request has merge conflicts that need to be resolved. It cannot be imported to the ROCm/rocm-systems repo automatically.

jrmadsen committed:
Update GOTCHA submodule

- public API for gotcha_init
- switch repo to ROCm/gotcha

jrmadsen committed:
Disable Perfetto logs

jrmadsen committed:
Update lib/output

- tool_counter_dimension_instance_info
- category table and category ids
- fix pmc info

jrmadsen committed:
Update SQL Schema

- fix summary views (use data views instead of rocpd_views)
- rocpd tables
  - new rocpd_info_category table
  - qualifier field in rocpd_info_pmc
  - new rocpd_info_address_range table
  - new rocpd_info_source_code table
  - new rocpd_info_pc table
  - remove call_stack/line_info JSON blob from rocpd_event
  - new rocpd_line_info table
  - new rocpd_call_stack_table
  - remove rocpd_metadata insert
- rocpd indexes
  - add new indexes
- data views
  - fixes regarding call_stack/line_info/category
  - misc cleanup

jrmadsen committed:
Update test_rocpd_data

- account for category changes

jrmadsen committed:
Update rocprofiler-sdk-tool

- update serialized records

jrmadsen committed:
rocpd interop GOTCHA updates

- fix issues wrapping dlopen/dlsym

jrmadsen committed:
Update rocpd python bindings

- rework perfetto implementation
  - category changes
- rework CSV implementation

jrmadsen committed:
rocpd filter support

jrmadsen committed:
Fix rocpd kernel dispatch CSV

jrmadsen committed:
rocpd merge fixes (csv.cpp + types.hpp + data_views.sql)

jrmadsen committed:
Linter and format fixes
